gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/oauth/OAuthPage.kt

```diff
@@ -63,23 +63,23 @@ internal fun OAuthPage(
     val lifecycle = LocalLifecycleOwner.current.lifecycle
     val uiState by viewModel.uiState.collectAsState()
 
-    LaunchedEffect(Unit) {
-        withContext(Dispatchers.Main.immediate) {
-            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                viewModel.actions.collect { action ->
-                    when (action) {
-                        OAuthAction.AuthorizationSuccess -> onAuthSuccess()
-                        OAuthAction.AuthorizationFailure -> onAuthError()
-                        OAuthAction.StartOAuth -> launchCustomTab(context, oAuthParams, email)
-                    }
-                }
-            }
+    val oAuthLauncher = rememberLauncherForActivityResult(GravatarOAuthResultContract()) { result ->
+        when (result) {
+            GravatarOAuthResult.DISMISSED -> Unit
+            is GravatarOAuthResult.TOKEN -> viewModel.tokenReceived(email, result.token)
+            GravatarOAuthResult.ERROR -> onAuthError()
         }
     }
 
+    // Kept for backwards-compatibility.
+    // This will be removed in the future and the GravatarOAuthActivity should be used instead.
     if (activity != null) {
         DisposableEffect(Unit) {
             val listener = Consumer<Intent> { newIntent ->
+                Log.w(
+                    "QuickEditor",
+                    "Setup GravatarOAuthActivity in your AndroidManifest.xml to handle the OAuth flow.",
```

Comment:
It works, but I'd be a little more explicit:

```suggestion
                    "GRAVATAR SDK WARNING: You're using a deprecated version of the Gravatar QuickEditor OAuth flow. Set up GravatarOAuthActivity in your AndroidManifest.xml to handle the OAuth flow and to remove this warning message. ",
```

Just a suggestion, feel free to improve this message.

Reply:
Looks good to me! A few UPPER CASE characters as a warning will help! :D